Add draft documentation with MD format (#20)
pypsa-meets-earth/pypsa-earth-status: doc/community/contributing.md

You can provide reference data, share suggestions and insights, help fellow modellers asking for assistance in [Discord](https://discord.gg/AnuJBk23FU) Support channel, contribute into outreach activities and events, and fund projects.

Feel free to use [this form](https://docs.google.com/forms/d/1udHf6W34YI0UNg3iwQs_-oeKsyj-dzJOtETZ_-RSUhw/edit) to share
your insights and check out our [website](https://pypsa-meets-earth.github.io)
for more details on ways to engage.

## Join us and get involved

Any person/group is welcome to join us. Be it research leader, researcher, undergraduate, or industry professional.
A simple way to explore opportunities for collaboration is to join our meetings. All of them are **OPEN**.

* PyPSA-Earth-Status periodic meeting: Every two weeks on Monday at 16:00 CET/CEST. If interested in joining, we can add you to the calendar invite; just reach out on [Discord](https://discord.gg/AnuJBk23FU) or via email to any of the core developers.
* [List of initiative meetings](https://pypsa-earth.readthedocs.io/en/latest/#get-involved)

* **Discord**
  * Chat with the community, team up on features, exchange with developers, code in voice channels

>   * [Discord invitation link](https://discord.gg/AnuJBk23FU)

Author: good point! I've rechecked the entire section. After testing, to have it recognized, the key seems to have 4 spaces

# How to Contribute

Contributions are welcome, and they are greatly appreciated!
Every little bit helps, and you always earn credits.

You can contribute on the code side in many ways:

* submit feedback,
* add new features,
* report bugs,
* fix bugs,
* implement a new cluster/cloud computation backend,
* write documentation

## Code

### Linting and pre-commit

> For every code contribution you should run [pre-commit](https://pre-commit.com/index.html).

Reviewer: Completely agree that it's much nicer to avoid pre-commit commits on the server but could it probably make sense to smoothen the tone a bit? Also, I think it's worth to mention that the pre-commit is run in an automated way. Adding an example (feel free to disregard),
```suggestion
For every code contribution we recommend to enable pre-commit running locally installing and activating [pre-commit](https://pre-commit.com/index.html).
```

Author: Revising slightly the text, making the sentence passive. It is also convenient for the user as having periodic diverging commit histories between local and github can be annoying

This will lint, format and check your code contributions against our guidelines
(e.g. we use [Black](https://github.com/psf/black) as code style
and aim for [REUSE compliance](https://reuse.software/)):

1. Installation `conda install -c conda-forge pre-commit` or `pip install pre-commit`

> 2. Usage:

Reviewer: I guess list formatting may need fixing:

```suggestion
2. Usage:

```

   * To automatically activate `pre-commit` on every `git commit`: Run `pre-commit install`
   * To manually run it: `pre-commit run --all`

## Documentation

### How to docs?

We add the code documentation along the way.
It might seem time-consuming and inefficient, but that's not really true anymore!
Documenting with great tools makes life much easier for YOU and YOUR COLLABORATORS and speeds up the overall process.
Using [MkDocs](https://www.mkdocs.org/) and its plugin
[mkdocstrings](https://mkdocstrings.github.io/) we document in our

> code scripts which then will automatically generate the documentation you might see here.

Reviewer: I'd suggest to shift a bit a focus from magnificent properties of `MkdDocs` (which are certainly worth proper acknowledgement) to the need of the project. For example:
```suggestion
We add the code documentation in a dynamic way along with the growth of the project. 
The documenting tools we are using [MkDocs](https://www.mkdocs.org/) and [mkdocstrings](https://mkdocstrings.github.io/) make documenting easy. Feel free to check it yourself by
starting to contribute. Every typo counts!
```

Thank you Eric Holscher & team for your wonderful *Readthedocs* open source project.
You can find an emotional speech by Eric [here](https://www.youtube.com/watch?v=U6ueKExLzSY).

> 

Author: No, dropping

```bash
.../pypsa-earth-status % conda create --name pypsa-earth-docs python
```

```bash
.../pypsa-earth-status % conda activate pypsa-earth-docs
```

Next, install the packages specified in `doc/requirements.txt` using `pip`:

```bash
.../pypsa-earth-status % pip install -r doc/requirements.txt
```

Once installation is completed, the following commands allow you to create the documentation locally:

```bash
.../pypsa-earth-status (pypsa-earth-docs) % cd doc
```

```bash
.../pypsa-earth-status/doc (pypsa-earth-docs) % mkdocs serve
```

This will start a local server, usually at `http://127.0.0.1:8000/`.
VScode provides a so called Liveserver extension such that the html file can be opened locally on your computer.

The documentation is built automatically by the CI for every pull request. The documentation is hosted on [ReadTheDocs](https://pypsa-earth-status.readthedocs.io/en/latest/).

## No-Code

Instead of contributing code there are alternatives to support the PyPSA-Earth-Status goals.
You can fund projects, supervise people, provide reference data and support us with outreach activities or events.

> Check out our [website](https://pypsa-meets-earth.github.io) for more details.

Reviewer: Trying to re-arrange the suggested contributions according to my understanding of their relevance (feel free to disagree!):

```suggestion
You can provide reference data and share your professional insights relevant for
validation, help fellow modellers asking for assistance in [Discord](https://discord.gg/AnuJBk23FU) Support channel,
contribute into outreach activities and events. Feel free to use [this form](https://docs.google.com/forms/d/1udHf6W34YI0UNg3iwQs_-oeKsyj-dzJOtETZ_-RSUhw/edit) to share
your insights and check out our [website](https://pypsa-meets-earth.github.io)
for more details on ways to engage.
```
